src/container/CandidateComponent.js

```diff
@@ -0,0 +1,106 @@
+import { useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { sortedCandidates } from "../redux/actions/candidateActions";
+import { useLocation } from "react-router";
+import { SORT_BY } from "./const";
+
+const CandidateComponent = () => {
+  const [searchBy, setSearchBy] = useState("");
+  const location = useLocation();
+  const filterParam = new URLSearchParams(location.search).get("search");
+  const dispatch = useDispatch();
+  const candidates = useSelector((state) => state.candidateReducer);
+  const renderList = (input) => {
+    if (input)
+      return input.map((candidate) => {
+        const {
+          id,
+          name,
+          email,
+          birth_date,
+          year_of_experience,
+          position_applied,
+          application_date,
+          status,
+        } = candidate;
+        return (
+          <tr>
+            <td key={id} data-label="Name">
+              {name}
+            </td>
+            <td data-label="Email">{email}</td>
+            <td data-label="DOB">{birth_date}</td>
+            <td data-lebel="Years of Experience">{year_of_experience}</td>
+            <td data-label="Position Applied">{position_applied}</td>
+            <td data-label="Application Date">{application_date}</td>
+            <td data-lbel="Status">{status}</td>
+          </tr>
+        );
+      });
+  };
+
+  const search = () => {
+    if (searchBy != "") {
+      return candidates.filter(
+        (el) => el.name.toLowerCase().indexOf(searchBy) > -1
+      );
+    } else {
+      return candidates;
+    }
+  };
+
+  const sortBy = (type) => {
+    if (type != "") {
+      dispatch(sortedCandidates({ candidates: candidates, sortType: type }));
+    } else {
+      return candidates;
+    }
+  };
+
+  return (
+    <div className="ui main text container">
+      <h2> Candidate Component </h2>
+      <div>
+        <input
```

Comment:
Not suggesting for current project, but always add debouncing whenever dealing with searching or making an API call in case of <input/> textbox.